openhgnn/models/MAGNN.py

```diff
@@ -433,56 +476,98 @@ def mp_instance_sampler(g, metapath_list):
 
     Notes
     -----
-    Please make sure that the metapath in metapath_list are all symmetric
+    Please make sure that the metapath in metapath_list are all symmetric and bidirectional. If not bidirectional,
+    please make sure that the leftmost element in a metapath in the dst node type.
+
+    We'd store the metapath instances in the disk after one metapath instances sampling and next time the
+    metapath instances will be extracted directly from the disk if they exists.
```

Comment:
```
This function caches all the metapath instances on the disk at path <fill in the path here>.  It will build the cache if the file does not exist.
```
That's also one reason why I prefer putting the path in the arguments.